### PR TITLE
TW-1183: Missing tag highlight on messages included url and tag name

### PR DESCRIPTION
**Pull request by hieutbui**
## Ticket
- #1183 

## Root cause
- The current message content for preview link message don't handle case tag name.

## Solution
- Use `HtmlMessage` widget instead.

## Resolved

https://github.com/linagora/twake-on-matrix/assets/80142234/b2adb7ba-788d-4f41-84ce-29f9b3e80e9c

### CODE REVIEW

File: lib/pages/chat/events/message_content.dart [hunk 1]
```diff
@@ -144,33 +144,13 @@ class MessageContent extends StatelessWidget
...
-                    maintainSize: true,
-                    maintainAnimation: true,
-                    maintainState: true,
-                    child: endOfBubbleWidget,
-                  ),
+                child: MessageTextContent(
+                  event: event,
+                  textColor: textColor,
+                  fontSize: fontSize,
+                  endOfBubbleWidget: endOfBubbleWidget,
```
Comment (hoangdat) [inline lib/pages/chat/events/message_content.dart:153]: why we need to change widget for emoji?

Comment (hoangdat) [reply lib/pages/chat/events/message_content.dart:153]: ah, got your idea. so we should change the name of widget

File: lib/widgets/twake_components/twake_preview_link/twake_link_preview.dart [hunk 2]
```diff
@@ -24,6 +24,7 @@ class TwakeLinkPreview extends StatefulWidget {
   final double? fontSize;
   final bool ownMessage;
   final TextSpanBuilder? textSpanBuilder;
+  final Widget? messageContentWidget;
```
Comment (hoangdat) [inline lib/widgets/twake_components/twake_preview_link/twake_link_preview.dart:27]:
```suggestion
  final Widget? formattedTextContextWidget;
```

Comment (hoangdat) [reply lib/widgets/twake_components/twake_preview_link/twake_link_preview.dart:27]: Should move logic of detecting formatted text into this widget

**Review by Te-Z — APPROVED** [review]

File: lib/pages/chat/events/message_content.dart (hunk 1 above)
Comment (hieutbui) [reply lib/pages/chat/events/message_content.dart]: done

**Comment by sherlockvn** [comment]
Hi i can't tag anyone in any room. Can you have a look ?

https://github.com/linagora/twake-on-matrix/assets/43041967/37077da5-2634-4e4f-a3d0-67952615a04b

**Review by hoangdat — APPROVED** [review]

**Pull request merged**